fw/hpack.c

```diff
@@ -927,6 +927,48 @@ tfw_hpack_add_index(TfwHPackDTbl *__restrict tbl,
 	return 0;
 }
 
+/*
+ * When header name and/or value is taken from either
+ * static or dynamic table, check if entry tag is valid
+ * (i.e. this header is valid for request)
+ * Valid pseudo-headers for requests/responses are defined
+ * in RFC 7540 8.1.2.3/8.1.2.4
+ */
+static bool
+tfw_hpack_entry_tag_valid(unsigned int entry_tag)
+{
+	bool is_valid = false;
+
+	switch (entry_tag) {
+	case TFW_TAG_HDR_H2_METHOD:
+	case TFW_TAG_HDR_H2_SCHEME:
+	case TFW_TAG_HDR_H2_AUTHORITY:
+	case TFW_TAG_HDR_H2_PATH:
+	case TFW_TAG_HDR_ACCEPT:
+	case TFW_TAG_HDR_AUTHORIZATION:
+	case TFW_TAG_HDR_CACHE_CONTROL:
+	case TFW_TAG_HDR_CONTENT_LENGTH:
+	case TFW_TAG_HDR_CONTENT_TYPE:
+	case TFW_TAG_HDR_COOKIE:
+	case TFW_TAG_HDR_FORWARDED:
+	case TFW_TAG_HDR_HOST:
+	case TFW_TAG_HDR_IF_MODIFIED_SINCE:
+	case TFW_TAG_HDR_IF_NONE_MATCH:
+	case TFW_TAG_HDR_PRAGMA:
+	case TFW_TAG_HDR_REFERER:
+	case TFW_TAG_HDR_X_FORWARDED_FOR:
+	case TFW_TAG_HDR_USER_AGENT:
+	case TFW_TAG_HDR_TRANSFER_ENCODING:
+	case TFW_TAG_HDR_RAW:
+		is_valid = true;
+		break;
+	default:
+		T_DBG3("%s: HPACK entry tag (%d) is not valid for HTTP/2 req\n",
+			__func__, entry_tag);
+	}
+	return is_valid;
+}
```

Comment:
`entity->tag` is used to reference HTTP "special" headers (see `tfw_http_hdr_t` in http.h) and set `parser->_hdr_tag`, which is used by `tfw_http_msg_hdr_close()` for direct reusage of the static table entries for our special hesders table (see `tfw_hpack_hdr_set()`). Thus, `tfw_hpack_entry_tag_valid()` must validate all the used `tfw_http_hdr_t` constants for `tag` value in `static_table` from hapck.c. 

In this sense `set-cookie` should have `TFW_HTTP_HDR_SET_COOKIE` tag in `static_table`. Please also add test for this and maybe other headers if I missed some of them.